ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/RandomRule.java

```diff
@@ -48,17 +48,18 @@ public Server choose(ILoadBalancer lb, Object key) {
             List<Server> upList = lb.getReachableServers();
             List<Server> allList = lb.getAllServers();
 
+            int upCount = upList.size();
             int serverCount = allList.size();
-            if (serverCount == 0) {
+            if (serverCount == 0 || upCount == 0) {

```

Comment:
Check upCount isn't sufficient ?

Reply:
If the server that is reachable is down, then the upCount is 0